src/components/layout/Workflow/Workflow.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+
+import { useGlobalState } from '../../../state/contexts/GlobalState';
+import { useRegistrationState } from '../../../state/contexts/RegistrationState';
+import { WorkflowProps } from '../../../types';
+import WorkflowButtons from '../../inputs/buttons/WorkflowButtons/WorkflowButtons';
+
+function Workflow({ stages }: WorkflowProps) {
+  const [{ stage, isFirstStage, isLastStage }, dispatchRegisterState] =
+    useRegistrationState();
+  const [{ walletAddress, isSearching }, dispatchGlobalState] =
+    useGlobalState();
+
+  const [nextCondition, setNextCondition] = useState<boolean>(false);
+  const [backCondition, setBackCondition] = useState<boolean>(false);
+
+  useEffect(() => {
+    // eslint-disable-next-line
+    Object.entries(stages).map(([key, value], index) => {
+      if (index === stage) {
+        setNextCondition(value.nextCondition);
+        setBackCondition(value.backCondition);
+        return;
+      }
+    });
+  }, [stage, stages]);
+
+  function showConnectWallet() {
+    dispatchGlobalState({
+      type: 'setShowConnectWallet',
+      payload: true,
+    });
+  }
+
+  return (
+    <>
+      {/* eslint-disable-next-line */}
+      {Object.entries(stages).map(([key, value], index) => {
+        if (index === stage) {
+          return value.component;
```

Comment:
<img width="947" alt="image" src="https://user-images.githubusercontent.com/12800001/208149381-fae34bcc-735f-4be4-b555-91b48ef32825.png">
```suggestion
          return React.clone(value.component, {
              key
          })
```